src/boot/efi/meson.build

```diff
@@ -10,17 +10,26 @@ if not get_option('efi') or get_option('gnu-efi') == 'false'
         subdir_done()
 endif
 
```

Comment:
> But arch for example has no 32bit port anymore. There you'd want to have a way to build ia32 without having to do a full 32bit cross build just so you can toss out most of it. You'd need to provide a minimum of a lot of 32bit dependencies just so meson could run successfully, even when you just want to issue the build commands for the ia32 efi binaries to speed things up.

But that sounds like a workaround on a deliberate choice that arch made, that they don't want to support 32bit anymore. Shoehoring a 32bit build into a pretend-64bit build is a hack, no?

anyway, @keszybz what's your take on this?

Reply:
It is perfectly fine for a distro to drop 32bit kernel/userspace. But that still means that you need to provide the necessary tools to boot your 64bit OS. And there are 64bit systems that have been crippled with a 32bit EFI firmware and arch should be able to easily support this. (I feel by your logic it should drop i386-pc grub too.)

Also, I highly doubt dpkg even allows cross-arch package installs (lib32-* packages are still amd64 packages). What seem to suggest is pulling ia32 binaries out of the i386 dpkg and stuff it into the amd64 dpkg. That just sounds like a terrible idea.

@bluca you seem to be involved in debian, what's your take on this?

Reply:
So yeah, what @bluca describes how Ubuntu does it makes sense to me: drop i386 mostly, but do i386 builds still for a select set of packages.

Reply:
Debian and Ubuntu both have multiarch support and allow installing foo:i386 package on an amd64 system for example. (which results in installing or coinstalling foo_1.23-3_i386.deb and foo_1.23-3_amd64.deb on a single system). However we have been doing some interesting packaging to build package named foo-efi-ia32 as either _amd64 or _all whilst actually containing the EFI 32-bit binaries. So building/installing stuff in general has solutions.

In terms of hardware, I know that obsolete Macbooks Intel Core Duo are 64bit CPU but 32bit efi (note, not Core 2 Duo). And there were Intel MinnowBoards that had 64-bit cpu, had both 32 bit & 64 bit EFI firmware available, however by default out of the factory they shipped with the 32 bit firmware.

I am not aware of any other popular / large systems that have 32bit EFI firmware yet 64bit CPU. My understanding is that no other SKU/vendor repeated that mistake since. And all platforms since ship with matching EFI firmware and CPU, being 64-bit one.

Minnow board users should reflash into 64bit EFI firmware. The core duo macbook I had in University, it is long obsolete and not supported in terms of power efficiency CPU disk space etc. They can continue to use whatever they have on them, and shouldn't need to support or install new OS on them. Their batteries should be long dead too, and impossible to replace.

I am interested to know if there are any other SKUs that are being manufactured and if anybody supports them that ship ia32 efi firmware. I.e. stuff released in the last 10 years, i.e. from 2013 onwards. And has received Spectre Meltdown mitigations.

Reply:
Yeap, after my post above I got pointed at machines sold in 2015 with such setup. Not flagships or anything, but yeah, they were still made and sold less than 8 years ago.

Reply:
Code review is def not the place for technical support, let alone when you bring in problems that are obviously distro-specific.

Reply:
@mirh I didn't ask for any tech support. I was just relaying some info about a device that uses a 64-bit CPU and a 32-bit UEFI. The rest was a little back story in case someone was still working on this so they would know a little about what I did.